### PR TITLE
fix(world): check namespace exists before balance transfer [L-03]

### PR DESCRIPTION
**Pull request by yonadaaa**

### CODE REVIEW

File: packages/world/src/modules/core/implementations/BalanceTransferSystem.sol [hunk 1]
```diff
@@ -37,6 +38,9 @@ contract BalanceTransferSystem is System, IWorldErrors {
       revert World_InvalidResourceType(RESOURCE_NAMESPACE, toNamespaceId, toNamespaceId.toString());
     }
 
+    // Require the namespace to exist
+    if (!ResourceIds.getExists(toNamespaceId)) revert World_ResourceNotFound(toNamespaceId, toNamespaceId.toString());
+
     // Require caller to have access to the namespace
     AccessControl.requireAccess(fromNamespaceId, _msgSender());
```
Comment (holic) [inline packages/world/src/modules/core/implementations/BalanceTransferSystem.sol:45]: I wonder how much gas we'd incur if we added this check in `requireAccess` instead?

**Review by holic — DISMISSED** [review]

File: packages/world/src/modules/core/implementations/BalanceTransferSystem.sol [hunk 2]
```diff
@@ -37,6 +38,9 @@ contract BalanceTransferSystem is System, IWorldErrors {
       revert World_InvalidResourceType(RESOURCE_NAMESPACE, toNamespaceId, toNamespaceId.toString());
     }
 
+    // Require the namespace to exist
+    if (!ResourceIds.getExists(toNamespaceId)) revert World_ResourceNotFound(toNamespaceId, toNamespaceId.toString());
```
Comment (alvrs) [inline packages/world/src/modules/core/implementations/BalanceTransferSystem.sol:42]: style nit (wonder if we can configure a linter to do this for consistency)
```suggestion
    if (!ResourceIds.getExists(toNamespaceId)) {
      revert World_ResourceNotFound(toNamespaceId, toNamespaceId.toString());
    }
```

**Comment by alvrs** [comment]
just added a `requireExistence` helper to `AccessControl` in https://github.com/latticexyz/mud/pull/2007, we can use that for this check too!

**Comment by yonadaaa** [comment]
This PR is failing integration tests because `CoreSystem` is too big to deploy. This should be fixed in https://github.com/latticexyz/mud/pull/2046.

**Comment by yonadaaa** [comment]
Using the `requireExistence` helper got this under the size limit!

File: .changeset/seven-carpets-develop.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@latticexyz/world": patch
+---
+
+Require that namespaces exist before transferring ETH to them.
```
Comment (holic) [inline .changeset/seven-carpets-develop.md:5]:
```suggestion
Namespace balances can no longer be transferred to non-existent namespaces.
```
reworded for consistent tense with other changesets and to be a bit more specific about namespace balances

**Review by holic — APPROVED** [review]

**Pull request merged**